stdlib/src/builtin/int.mojo

```diff
@@ -1099,6 +1148,51 @@ struct Int(
     # Methods
     # ===-------------------------------------------------------------------===#
 
+    fn mod_inverse(self, m: Self) -> Self:
+        """Compute the modular multiplicative inverse of `self` modulo `m`.
+
+        This method uses the extended Euclidean algorithm to find the modular
+        multiplicative inverse.
+        The result `x` satisfies (`self` * `x`) % `m` == 1.
+
+        Args:
+            m: The modulus. Must be greater than 1.
+
+        Returns:
+            The modular multiplicative inverse of `self` modulo `m`, if it exists.
+            Returns 0 if no inverse exists or if an error occurs.
+
+        Note:
+            - If `self` and `m` are not coprime, no modular inverse exists,
+              and the method returns 0.
+        """
+        if m <= 1:
```

Comment:
In Python: 
- `pow(1, 1, 0)` raises `ValueError: pow() 3rd argument cannot be 0`
- `pow(0, 1, 1)` returns `0`. 
- `pow(0, 1, -1)` returns `0`. (definitely not valid math)
- `pow(50, 1, -50)` returns `0`. (definitely not valid math)

I think the mod inverse of 0 doesn't exist either since any number mod by itself is 0. But I think we should keep to Pyhton's behavior.

If we take a `UInt` into this function it might also lead to -50 for example being cast to its positive equivalent (18446744073709551566) and giving different results than Python which is sad.

And as for 0 that raises in Python,  I personally have an inclination towards aborting for cases of undefined Math to not mark the function as `raises` instead of raising. But it needs to be decided by the community/Mojo team how we deal with these cases as a whole.